libs/lib-client/mapper/src/services/mapper-controller/mapper-controller-factory.service.ts

```diff
@@ -28,22 +27,33 @@ export class MapperControllerFactory {
     input: AttributeDescriptor[],
     output: AttributeDescriptor[],
     mappings: any,
-    functions: InstalledFunctionSchema[]
+    functions: InstalledFunctionSchema[],
+    makeSnippet: (nodes) => string,
```

Comment:
I wonder if we need an interface for this function? I see it is used in many places. Also can we type nodes?

```ts
type CreateSnippetFn = (nodes: type(??)) => string;
// ...
makeSnippet: CreateSnippetFn,
```